tests/test_compressors/quantized_compressors/test_asymmetric_decompression.py

```diff
@@ -0,0 +1,228 @@
+        
+        assert reconstructed_weights["layer1"]["weight"].shape == original_weights["layer1"].shape
+        assert reconstructed_weights["layer2"]["weight"].shape == original_weights["layer2"].shape
+        
+        new_model = SimpleModel()
+        new_model.layer1.weight.data = reconstructed_weights["layer1"]["weight"]
+        new_model.layer2.weight.data = reconstructed_weights["layer2"]["weight"]
+        
+        test_input = torch.randn(1, 512)
+        with torch.no_grad():
+            output = new_model(test_input)
+        
+        assert output.shape == (1, 128)
+        assert not torch.isnan(output).any()
+        assert not torch.isinf(output).any()
+
+
+@pytest.mark.parametrize("num_bits", [4, 8])
+def test_asymmetric_quantization_accuracy(num_bits):
+    """
+    Test that asymmetric quantization with zero-point preserves accuracy better
+    than symmetric quantization for biased weight distributions.
+    """
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        tmp_path = Path(tmp_dir)
+        
+        shape = (256, 512)
+        weights = torch.randn(shape) + 2.0
+        
+        quant_config = create_asymmetric_quant_config(
+            num_bits=num_bits,
+            strategy=QuantizationStrategy.GROUP,
+            group_size=128
+        )
+        
+        group_size = 128
+        num_groups = shape[1] // group_size
+        scale_shape = (shape[0], num_groups)
+        
+        scales = torch.rand(scale_shape) * 0.1
+        zero_points = torch.randint(-2**(num_bits-1), 2**(num_bits-1), scale_shape, dtype=torch.int8)
+        
+        state_dict = {
+            "layer.weight": weights,
+            "layer.weight_scale": scales,
+            "layer.weight_zero_point": zero_points,
+        }
+        
+        compressor = PackedQuantizationCompressor(config=quant_config)
+        quantized_modules_to_scheme = {"layer": quant_config.config_groups["group_1"]}
+        
+        compressed_state_dict = compressor.compress(
+            state_dict.copy(), names_to_scheme=quantized_modules_to_scheme
+        )
+        
+        save_file(compressed_state_dict, tmp_path / "model.safetensors")
+        
+        reconstructed_gen = compressor.decompress(
+            tmp_path, names_to_scheme=quantized_modules_to_scheme
+        )
+        
+        reconstructed = {}
+        for module_name, module_data in reconstructed_gen:
+            reconstructed[module_name] = module_data
+        
+        assert "layer" in reconstructed
+        assert "weight" in reconstructed["layer"]
+        assert reconstructed["layer"]["weight"].shape == shape
+        
+        decompressed_weights = reconstructed["layer"]["weight"]
+        assert not torch.isnan(decompressed_weights).any()
+        assert not torch.isinf(decompressed_weights).any()
+        
+        assert decompressed_weights.abs().max() < 100
+        assert decompressed_weights.abs().max() > 0.01
```

Comment:
Rather than checking abs-max, it would be better to assert the difference between original and reconstructed decompressed are within a given threshold. Maybe we could compare the stdev is within the stdev of two random matrices?
```suggestion
        threshold = torch.std(torch.rand(shape) - torch.rand(shape))
        assert torch.std(weights - decompressed_weights) < threshold
```